File: leetcode/p0532/Solution.java
class Solution {
    public int findPairs(int[] nums, int k) {
        int count = 0;
        Map<Integer, Integer> map = new TreeMap<>();
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        for (int num : map.keySet()) {
            if (map.containsKey(num + k)) {
                if (k != 0 || map.get(num) > 1) {
                    count++;
                }
            }    
        }
        return count;
    }
}

File: leetcode/p0532/Solution.kt
class Solution {
    fun findPairs(nums: IntArray, k: Int): Int {
        val map = TreeMap<Int, Int>()
        for (num in nums) {
            map[num] = (map[num] ?: 0) + 1
        }

        var count = 0
        for (num in map.keys) {
            if (map.containsKey(num + k)) {
                if (k != 0 || (map[num] ?: 0) > 1) {
                    count++
                }
            }
        }
        return count
    }
}
